relay-server/src/metrics_extraction/transactions.rs

```diff
@@ -199,10 +215,50 @@ fn keep_transaction_name(source: &TransactionSource) -> bool {
     }
 }
 
+/// Decide whether we want to keep the transaction name.
+/// High-cardinality sources are excluded to protect our metrics infrastructure.
+/// Note that this will produce a discrepancy between metrics and raw transaction data.
+fn get_transaction_name(
+    event: &Event,
+    accept_transaction_names: &AcceptTransactionNames,
+) -> Option<String> {
+    let original_transaction_name = match event.transaction.value() {
+        Some(name) => name,
+        None => {
+            return None;
+        }
+    };
+
+    let use_original_name = match accept_transaction_names {
+        AcceptTransactionNames::ClientBased => {
+            let sdk_name = event
+                .client_sdk
+                .value()
+                .and_then(|sdk| sdk.name.value())
+                .map(|s| s.as_str())
+                .unwrap_or_default();
+
+            (sdk_name != "sentry.javascript.react" && sdk_name != "sentry.javascript.browser")
+                || is_low_cardinality(event.get_transaction_source())
```

Comment:
This condition sets `use_original_name` to `true` if any other client sends `TransactionSource::Url`. However, we want URLs to always be false. The difference should only exist if the source is `Unknown`.